Patch-Source: https://gitlab.com/kicad/code/kicad/-/merge_requests/2791
--
From 16aefc553b8bd847e20a11b26af0678e7af74989 Mon Sep 17 00:00:00 2001
From: qu1ck <anlutsenko@gmail.com>
Date: Wed, 2 Sep 2026 02:56:39 -0700
Subject: [PATCH] Fix swig api iterators usage

SWIG 4.4.0 and later drops python 2 style .next() methods for SwigPyIterator
and only uses __next__() which are used by the python 3 next() builtin.

Fixes https://gitlab.com/kicad/code/kicad/-/work_items/25423
---
 pcbnew/python/swig/board.i | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pcbnew/python/swig/board.i b/pcbnew/python/swig/board.i
index 850bd9f39e4..e71b7268ef0 100644
--- a/pcbnew/python/swig/board.i
+++ b/pcbnew/python/swig/board.i
@@ -111,7 +111,7 @@ HANDLE_EXCEPTIONS(BOARD::TracksInNetBetweenPoints)
             it = self.iterator()
             try:
                 while True:
-                    item = it.next()  # throws StopIteration when iterator reached the end.
+                    item = next(it)  # throws StopIteration when iterator reached the end.
                     yield item.Cast()
             except StopIteration:
                 return
@@ -125,7 +125,7 @@ HANDLE_EXCEPTIONS(BOARD::TracksInNetBetweenPoints)
             it = self.iterator()
             try:
                 while True:
-                    item = it.next()  # throws StopIteration when iterator reached the end.
+                    item = next(it)  # throws StopIteration when iterator reached the end.
                     yield item.Cast()
             except StopIteration:
                 return
-- 
GitLab

